app/ui/src/app/customizations/api-connector/api-connector.reducer.ts

```diff
@@ -48,10 +61,69 @@ export function apiConnectorReducer(state = initialState, action: any): ApiConne
       };
     }
 
+    case ApiConnectorActions.CREATE: {
+      const createRequest = (action as ApiConnectorCreate).payload;
+      return {
+        ...state,
+        createRequest: { ...createRequest, isComplete: false, isRequested: true },
+        loading: true,
+        hasErrors: false,
+        errors: []
+      };
+    }
+
+    case ApiConnectorActions.CREATE_COMPLETE: {
+      const createRequest = (action as ApiConnectorCreateComplete).payload;
+      return {
+        ...state,
+        createRequest: { ...createRequest, isComplete: true },
+        loading: false,
+        hasErrors: false,
+        errors: []
+      };
+    }
+
+    case ApiConnectorActions.CREATE_FAIL: {
+      return {
+        ...state,
+        loading: false,
+        hasErrors: true,
+        errors: [action.payload]
+      };
+    }
+
+    case ApiConnectorActions.CREATE_CANCEL: {
+      return {
+        ...state,
+        createRequest: undefined,
+        loading: false,
+        hasErrors: false,
+        errors: []
+      };
+    }
+
+    case ApiConnectorActions.UPDATE_AUTH_SETTINGS: {
+      const configuredProperties = {
+        ...state.createRequest.configuredProperties,
+        ...action.payload
+      };
+      return {
+        ...state,
+        createRequest: {
+          ...state.createRequest,
+          configuredProperties,
+        },
+        loading: false,
+        hasErrors: false,
+        errors: []
+      };
+    }
+
     case ApiConnectorActions.FETCH: {
       return {
         ...state,
         loading: true,
+        loaded: false,
```

Comment:
We are not returning a default state in case there are no changes?

Reply:
Ah nevermind, I see it here: https://github.com/deeleman/syndesis/blob/0c76af5ce20716b0253338c02615f194a77f34fd/app/ui/src/app/customizations/api-connector/api-connector.reducer.ts#L155 👍

Reply:
> We are not returning a default state in case there are no changes?

Certainly we do! But changes is not the right term here, Rachel. Our reducer handles ACTIONS. Therefore, we do return a default state (which is the current state snapshot) when the action that executes this reducer is not intercepted by any action handler in the switch/case block.

Reply:
Correct, I'm terrible with terminology.